### PR TITLE
Remove bootstrap_height from Leader Scheduler

### PR DESCRIPTION
**Pull request by mvines**
`leader_scheduler` no longer needs the implementation concept of bootstrap_height, which removes a bunch of code and special cases from the leader scheduler. 

`bank::process_genesis_block` now configures a voting account for the bootstrap leader such that the leader scheduler need not care if it's scheduling slot 0 or slot N.  

TODO:
* [x] Fix up all the busted tests
* [x] Rebase to the tip of master
* [x] Clean up hacks

### CODE REVIEW

**Comment by mvines** [comment]
Update: I'm down to a couple broken multinode tests here.

File: src/fullnode.rs [hunk 1]
```diff
@@ -382,6 +425,38 @@ impl Fullnode {
...
+                }
+                Some(transition) => {
+                    debug!("role_transition complete: {:?}", transition);
+                    if let Some(ref rotation_notifier) = rotation_notifier {
+                        rotation_notifier.send(transition).unwrap();
+                    }
+                }
+            };
+        });
+        move || {
```
Comment (rob-solana) [inline src/fullnode.rs]: this is clever, what happens on drop() of the closure?

Comment (mvines) [reply src/fullnode.rs]: If/when somebody else sets `exit` to true eventually the thread above should panic at `sender.send(true).expect("Unable to signal exit");`, which seams fine cause it's done working at that point anyway

**Comment by mvines** [comment]
image

**Comment by carllin** [comment]
https://www.youtube.com/watch?v=Bkq1PAyGuZY

**Pull request merged**